anitya/lib/model.py

```diff
@@ -369,6 +302,26 @@ class Project(BASE):
                             name="UNIQ_PROJECT_NAME_PER_ECOSYSTEM"),
     )
 
+    @validates('backend')
+    def validate_backend(self, key, value):
```

Comment:
We may have to revisit this logic later if we ever remove any plugins (such that a backend or ecosystem is referenced in the DB without a runtime plugin available), but it seems fine as long as we're only adding new plugins, rather than removing or renaming existing ones.